use-cases/model-fine-tuning-pipeline/data-preparation/gemma-it/src/openai_credentials_refresher.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any
+
+import os
+import json
+import logging
+import logging.config
+import google.auth
+import google.auth.transport.requests
+import openai
+
+import tenacity
+
+# Configure logging
+logging.config.fileConfig("logging.conf")
+
+logger = logging.getLogger("openaiauth")
+
+if "LOG_LEVEL" in os.environ:
+    new_log_level = os.environ["LOG_LEVEL"].upper()
+    logger.info(
+        f"Log level set to '{new_log_level}' via LOG_LEVEL environment variable"
+    )
+    logging.getLogger().setLevel(new_log_level)
+    logger.setLevel(new_log_level)
+
+from tenacity import retry, stop_after_attempt, wait_random
+
+
+class OpenAICredentialsRefresher:
+    def __init__(self, **kwargs: Any) -> None:
```

Comment:
Ideally this should use the same `processing` logger unless you have a specific reason for it to have it's own logger.

Reply:
I actually tried switching `processing` and it broke logging for dataprep and the refresher. both with only getLogger and with the full init block

Reply:
Did you pass the logger into the class on initialization? You shouldn't be creating another logger instances. You would just pass the logger to the class and set it, then use the self.logger.

Reply:
thanks, refactored and updated